extensions/wikia/AssetsManager/builders/AssetsManagerSassBuilder.class.php

```diff
@@ -4,40 +4,54 @@
  * @author Inez Korczyński <korczynski@gmail.com>
  * @author Piotr Bablok <piotr.bablok@gmail.com>
  */
-
 class AssetsManagerSassBuilder extends AssetsManagerBaseBuilder {
 	const CACHE_VERSION = 2;
 
-	public function __construct(WebRequest $request) {
+	/**
+	 * @var bool $exists Whether the file we're trying to load exists
+	 */
+	protected $exists = true;
+
+	public function __construct( WebRequest $request ) {
 		global $IP;
-		parent::__construct($request);
+		parent::__construct( $request );
 
 		// TODO: the background image shouldn't be passed as the url - we should pass a File reference and derive ourselves
-		if (isset($this->mParams['background-image']) && VignetteRequest::isVignetteUrl($this->mParams['background-image']) && isset($this->mParams['path-prefix'])) {
-			$connector = strpos($this->mParams['background-image'], '?') === false ? '?' : '&';
+		if ( isset( $this->mParams['background-image'] ) && VignetteRequest::isVignetteUrl( $this->mParams['background-image'] ) && isset( $this->mParams['path-prefix'] ) ) {
+			$connector = strpos( $this->mParams['background-image'], '?' ) === false ? '?' : '&';
 			$this->mParams['background-image'] .= "{$connector}path-prefix={$this->mParams['path-prefix']}";
 		}
 
-		if (strpos($this->mOid, '..') !== false) {
-			throw new Exception('File path must not contain \'..\'.');
+		if ( strpos( $this->mOid, '..' ) !== false ) {
+			throw new Exception( 'File path must not contain \'..\'.' );
 		}
 
-		if (!endsWith($this->mOid, '.scss', false)) {
-			throw new Exception('Requested file must be .scss.');
+		if ( !endsWith( $this->mOid, '.scss', false ) ) {
+			throw new Exception( 'Requested file must be .scss.' );
 		}
 
 		//remove slashes at the beginning of the string, we need a pure relative path to open the file
 		$this->mOid = preg_replace( '/^[\/]+/', '', $this->mOid );
 
 		if ( !file_exists( "{$IP}/{$this->mOid}" ) ) {
-			throw new Exception("File {$this->mOid} does not exist!");
+			// SUS-399: allow SCSS processing to continue, log the error
+			$this->exists = false;
+			Wikia\Logger\WikiaLogger::instance()->info( 'Nonexistent SCSS file requested', [
+				'oid' => $this->mOid,
+			] );
 		}
 		$this->mContentType = AssetsManager::TYPE_CSS;
 	}
 
 	public function getContent( $processingTimeStart = null ) {
 		global $IP, $wgEnableSASSSourceMaps;
-		wfProfileIn(__METHOD__);
+		wfProfileIn( __METHOD__ );
+
+		// SUS-399: Abort early if we received a request for a file that no longer exists
+		if ( !$this->exists ) {
+			wfProfileOut( __METHOD__ );
+			return '';
+		}
```

Comment:
Will such requests still be handled correctly - i.e. return [`HTTP 404`](http://slot1.images4.wikia.nocookie.net/__am/1469493438/sasses/background-dynamic%3Dfalse%26background-image%3Dhttp%253A%252F%252Fvignette3.wikia.nocookie.net%252Fpoznan%252Fimages%252F5%252F50%252FWiki-background%252Frevision%252Flatest%253Fcb%253D20150517124735%2526path-prefix%253Dpl%26background-image-height%3D288%26background-image-width%3D300%26color-body%3D%2523f3e2d0%26color-body-middle%3D%2523f3e2d0%26color-buttons%3D%2523006cb0%26color-header%3D%25233a5766%26color-links%3D%2523006cb0%26color-page%3D%2523f8f4f0%26oasisTypography%3D1%26page-opacity%3D92%26widthType%3D0/skins/oasis/css/oasis.scss,extensions/wikia/Recirculation/styles/recirculation.scss,extensions/wikia/PortableInfobox/styles/PortableInfobox.scss,extensions/wikia/VisualEditorTourExperiment/styles/VisualEditorTourExperimentInit.scss,extensions/wikia/GlobalShortcuts/styles/GlobalShortcutsHelp.scss,extensions/wikia/GlobalShortcuts/styles/GlobalShortcutsSearch.scss,extensions/wikia/SpecialPromote/css/SpecialPromote.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigation.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationSearch.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationAccountNavigation.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationUserLoginDropdown.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationNotifications.scss,extensions/wikia/GlobalNavigation/styles/GlobalNavigationInverse.scss,extensions/wikia/GlobalNavigation/styles/GoogleSearchAPI.scss,skins/shared/styles/transparent-out.scss,extensions/wikia/PageShare/styles/PageShare.scss,extensions/wikia/GlobalFooter/styles/GlobalFooter.scss,extensions/wikia/Qualaroo/css/Qualaroo.scass)?